### PR TITLE
PSP: clarify the difference between the case when pod is being created by a user and a service account

### PR DESCRIPTION
**Pull request by php-coder**
This partially addresses https://github.com/kubernetes/kubernetes/issues/49024

PTAL @pweil- @liggitt @simo5 @strowi

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4423)

### CODE REVIEW

File: docs/concepts/policy/pod-security-policy.md [hunk 1]
```diff
@@ -206,6 +206,29 @@ following
...
+In Kubernetes 1.5 and newer, you can use PodSecurityPolicy to control access to
+privileged containers based on user role and groups. Access to different
+PodSecurityPolicy objects can be controlled via authorization.
+
+Note that [Controller Manager](/docs/admin/kube-controller-manager/) must be run
+against [the secured API port](/docs/admin/accessing-the-api/), and must not
+have superuser permissions. Otherwise requests would bypass authentication and
```
Comment (php-coder) [inline docs/concepts/policy/pod-security-policy.md]: To be honest, I don't understand "must not have superuser permissions" requirement.

@pweil- @liggitt do you know what does it mean?

Comment (liggitt) [inline docs/concepts/policy/pod-security-policy.md]: the controller manager runs controller loops that create pods based on replicasets, daemonsets, etc.

The PSP admission plugin will check those pods as they are created, based on the PSP permissions of the creating user (in this case, the controller loop) and the service account on the pod spec.

If the controller has superuser powers (can do * on *), then all PSP policies will be accessible to pods created by the controller, so anyone that could create a replicaset could make the controller manager create them the most powerful pod allowed by any PSP.

**Comment by strowi** [comment]
thx, that clarifies a little bit for me. But there is another weird behaviour that i noticed. Not sure if this belongs to psp-documentation, or a new bug?

Using the following policies with the examples, a restricted user can create a daemonset without specifying a serviceaccount and won't be restricted (e.g. use  ports < 3000, privileged-mode, etc).
Does Kubernetes only check for the existence of keys or the value too?

```
apiVersion: extensions/v1beta1
kind: PodSecurityPolicy
metadata:
  name: privileged
spec:
  privileged: true
  runAsUser:
    rule: RunAsAny
  fsGroup:
    rule: RunAsAny
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    rule: RunAsAny
  hostIPC: true
  hostPID: true
  hostNetwork: true
  hostPorts:
  - min: 0
    max: 65535
  volumes:
  - '*'
---
apiVersion: extensions/v1beta1
kind: PodSecurityPolicy
metadata:
  name: restricted
spec:
  privileged: false
  runAsUser:
    rule: RunAsAny
  fsGroup:
    rule: RunAsAny
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    rule: RunAsAny
  hostIPC: false
  hostPID: false
  hostNetwork: false
  hostPorts:
  - min: 30000
    max: 65535
  volumes:
  - 'emptyDir'
  - 'gitRepo'
  - 'secret'
  - 'nfs'
  - 'persistentVolumeClaim'
  - 'downwardAPI'
  - 'configMap'
  - 'projected'
```

**Comment by php-coder** [comment]
> Not sure if this belongs to psp-documentation, or a new bug?

@strowi I have too few information to say something for sure:

- how do you run kubernetes?
- do you use `local-up-cluster.sh` script?
- which version of kubernetes do you use?
- how exactly do you create a pod? The command is important because if you run it against unsecure port, then you will have privileged pemissions
- could you inspect a pod and check what PSP has been applied? (`kubectl get pod <pod> -o yaml | grep psp`)

Also it's better to move the discussion to a better place.

**Comment by strowi** [comment]
@php-coder Sorry, mixed up the 2 threads..:

- kubernetes 1.6.6 is run via coreos/matchbox
- kubectl is always run against the secured api-port with rbac enabled
- for the pods created by a deployment the privileged psp is being used, don't really know why, there is no reference to any privileged role etc.. anywhere.

I can happliy supply more info if needed, but don't want to spam..

Would love to move it, where is a better place ?;)

**Comment by php-coder** [comment]
@strowi Create an issue in this repo, because docs outdated anyway (I see that `ALLOW_ANY_TOKEN` was completely removed from the k8s some time ago, for example).

**Comment by strowi** [comment]
Done -> #4457

**Comment by php-coder** [comment]
@pweil- @liggitt PTAL

File: docs/concepts/policy/pod-security-policy.md [hunk 2]
```diff
@@ -206,6 +206,29 @@ following
...
+authorization modules, all PodSecurityPolicy objects would be allowed,
+and user will be able to create privileged containers.
+
+PodSecurityPolicy authorization uses the union of all policies available to the
+user creating the pod and
+[the service account specified on the pod](/docs/tasks/configure-pod-container/configure-service-account/).
+
+Access to given PSP policies for a user will be effective only when deploying
```
Comment (liggitt) [inline docs/concepts/policy/pod-security-policy.md]: s/deploying/creating/

File: docs/concepts/policy/pod-security-policy.md [hunk 3]
```diff
@@ -206,6 +206,29 @@ following
...
+Pods directly.
+
+For pods created on behalf of a user, in most cases by Controller Manager,
+access should be given to a service account under which related controller is
```
Comment (liggitt) [inline docs/concepts/policy/pod-security-policy.md]: should be given to the service account specified on the pod spec template

File: docs/concepts/policy/pod-security-policy.md (hunk 2 above)
Comment (php-coder) [inline docs/concepts/policy/pod-security-policy.md]: Thanks, it's fixed now.

File: docs/concepts/policy/pod-security-policy.md (hunk 3 above)
Comment (php-coder) [inline docs/concepts/policy/pod-security-policy.md]: Thanks! I've just fixed it too.

**Comment by php-coder** [comment]
@chenopis It's ready to merge. PTAL

**Comment by php-coder** [comment]
Resolved a conflict by update a link.

**Comment by chenopis** [comment]
Deploy preview ready!

Built with commit 425291751706f3ef8767104a96b556eb8ac9cad6

https://deploy-preview-4423--kubernetes-io-master-staging.netlify.com

**Pull request merged**